### PR TITLE
Bug/jwt

### PR DESCRIPTION
**Pull request by Slavchev03**

### CODE REVIEW

File: back-end/src/main/java/com/devminds/rentify/config/JwtService.java [hunk 1]
```diff
@@ -2,85 +2,107 @@
...
 public class JwtService {
+    private  Key signInKey;
 
-    private static final String SECRET_KEY = "secret";
-    private final PasswordEncoder passwordEncoder;
+    private static final long TOKEN_VALIDITY_DURATION = 2000000L;
 
-    public String extractUsername(String token) {
-        return extractClaim(token, Claims::getSubject);
+    private static final String secretKey = "21974a72b5ada4edad7a26657647bec67855af08ad10183d67f5cc4b0e31bce3" ;
```
Comment (danailivanov733) [inline back-end/src/main/java/com/devminds/rentify/config/JwtService.java:40]: move all the information about the token in environment variables

File: back-end/src/main/java/com/devminds/rentify/config/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -55,4 +55,51 @@ protected void doFilterInternal(
         }
         filterChain.doFilter(request, response);
     }
+
```
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/java/com/devminds/rentify/config/JwtAuthenticationFilter.java:58]: Remove commented code or move it to another branch to keep locally if it's important.

File: back-end/src/main/java/com/devminds/rentify/config/JwtService.java [hunk 3]
```diff
@@ -2,85 +2,107 @@
...
+            return extractClaim(token, Claims::getSubject);
 
-    public boolean isTokenExpired(String token) {
-        return extractExpiration(token).before(new Date());
     }
 
-    public Date extractExpiration(String token) {
-        return extractClaim(token, Claims::getExpiration);
+    private Claims extractAllClaims(String token) {
+            return Jwts.parserBuilder().setSigningKey(signInKey).build().parseClaimsJws(token).getBody();
```
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/java/com/devminds/rentify/config/JwtService.java:105]: parseClaimsJwt(...) check the token for expiration. You don't need the isTokenValid() or isTokenExpired() methods.

File: back-end/src/main/java/com/devminds/rentify/config/ApplicationConfig.java [hunk 4]
```diff
@@ -43,4 +47,13 @@ public AuthenticationManager authenticationManager(AuthenticationConfiguration c
         return config.getAuthenticationManager();
     }
 
+    @Bean
+    public Key signInKey() {
```
Comment (KalinVasilev-at-blankfactor) [inline back-end/src/main/java/com/devminds/rentify/config/ApplicationConfig.java:51]: You don't need to generate a random key here. Move the code for key generation from JwtService to this bean method. You could then pull in the key into the JwtService constructor.

**Review by danailivanov733 — APPROVED** [review]

**Pull request merged**